dbt_dry_run/test/node_runner/test_model_runner.py

```diff
@@ -78,6 +82,29 @@ def test_model_as_view_runs_create_view() -> None:
     assert node.compiled_sql in executed_sql
 
 
+def test_model_with_dbt_max_partition_variable_replaces_it() -> None:
```

Comment:
The max partition variable isn't replaced more just prepended to the query so a better test name:

`test_partitioned_incremental_model_declares_dbt_max_partition_variable`

Or something like that?